src/Test.UnitTests.Sarif.Multitool/RebaseUriCommandTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Multitool;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Multitool
+{
+    public class RebaseUriCommandTests : FileDiffingUnitTests
+    {
+        private static ResourceExtractor Extractor = new ResourceExtractor(typeof(RebaseUriCommandTests));
+
+        public RebaseUriCommandTests(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+        [Fact]
+        public void RebaseUriCommand_RebaseRunWithArtifacts()
+        {
+            string testFilePath = "RunWithArtifacts.sarif";
+
+            var options = new RebaseUriOptions
+            {
+                BasePath = @"C:\vs\src\2\s\",
+                BasePathToken = "SRCROOT",
+                Inline = true,
+                SarifOutputVersion = SarifVersion.Current,
+                PrettyPrint = true
+            };
+
+            RunAndValidateRebaseUriCommand(testFilePath, options);
+        }
+
+        private void RunAndValidateRebaseUriCommand(string testFilePath, RebaseUriOptions options)
+        {
+            string inputSarifLog = Extractor.GetResourceText($"RebaseUriCommand.Inputs.{testFilePath}");
+            string expectedOutput = Extractor.GetResourceText($"RebaseUriCommand.ExpectedOutputs.{testFilePath}");
+
+            string logFilePath = @"c:\logs\mylog.sarif";
+            StringBuilder transformedContents = new StringBuilder();
+
+            options.TargetFileSpecifiers = new string[] { logFilePath };
+
+            Mock<IFileSystem> mockFileSystem = ArrangeMockFileSystem(inputSarifLog, logFilePath, transformedContents);
+
+            var rebaseUriCommand = new RebaseUriCommand(mockFileSystem.Object);
+
+            int returnCode = rebaseUriCommand.Run(options);
+            string actualOutput = transformedContents.ToString();
+
+            returnCode.Should().Be(0);
+
+            Assert.True(
+                AreEquivalent<SarifLog>(actualOutput, expectedOutput),
+                GetErrorMessageWithDiffFileDetails(testFilePath, expectedOutput, actualOutput)
+                );
+        }
+
+        private static string GetErrorMessageWithDiffFileDetails(string testFilePath, string expectedOutput, string actualOutput)
+        {
+            // TODO: Rebaseline test output functionality is not integrated for these tests yet.
```

Comment:
>yet [](start = 92, length = 3)

Is this a big job? It seems like something that should be part of this PR. If you add it, it will be easy to review, just as the diff between this iteration and the next.

Reply:
i completely agree!! let me give it another shot today with a fresh mind!

---
In reply to: [298672159](https://github.com/microsoft/sarif-sdk/pull/1549#discussion_r298672159) [](ancestors = 298672159)

Reply:
done! all i need was to override the function which reads the expected output file. I did have to go change its access modifier though 

---
In reply to: [298692734](https://github.com/microsoft/sarif-sdk/pull/1549#discussion_r298692734) [](ancestors = 298692734,298672159)

Reply:
Very nice solution.

---
In reply to: [298764040](https://github.com/microsoft/sarif-sdk/pull/1549#discussion_r298764040) [](ancestors = 298764040,298692734,298672159)